tests/unit/test_event.py

```diff
@@ -0,0 +1,143 @@
+            'start_time': 1710506390376, 
+            'end_time': 1710506510376, 
+            'type': 'Notification', 
+            'subtype': 'HE3', 
+            'description': 'Failed Instrumentation Fire extinguisher test', 
+            'metadata': {'description': 'Failed Instrumentation Fire extinguisher test', 'notification': '22081904', 'notification_type': 'HE3', 'work_center': 'SUPV', 'work_order_number': '22082901'}, 
+            'asset_ids': [5534771458240925], 
+            'id': 5875589750425323, 
+            'last_updated_time': 1710506651232, 
+            'created_time': LATE_CREATED_TIME
+        }
+    ]
+
+@pytest.fixture
+def mock_get_events(mocker, event_data):
+    return_value = iter([EventList([Event(**data) for data in event_data])])
+    return mocker.patch("cdf_fabric_replicator.event.EventsReplicator.get_events", return_value=return_value)
+
+
+@pytest.fixture
+def mock_get_late_events(mocker, event_data):
+    return_value = iter([EventList([Event(**event_data[-1])])])
+    return mocker.patch("cdf_fabric_replicator.event.EventsReplicator.get_events", return_value=return_value)
+
+@pytest.fixture
+def mock_get_no_events(mocker):
+    return_value = iter([EventList([])])
+    return mocker.patch("cdf_fabric_replicator.event.EventsReplicator.get_events", return_value=return_value)
+
+@pytest.fixture
+def mock_set_event_state(mocker):
+    return mocker.patch("cdf_fabric_replicator.event.EventsReplicator.set_event_state", return_value=None)
+
+@pytest.fixture
+def mock_write_events_to_lakehouse_tables(mocker):
+    return mocker.patch("cdf_fabric_replicator.event.EventsReplicator.write_events_to_lakehouse_tables", return_value=None)
+
+def snake_to_camel(snake_str):
+    components = snake_str.split('_')
+    return components[0] + ''.join(x.title() for x in components[1:])
+
+def convert_dict_keys_to_camel_case(snake_case_dict):
+    return {snake_to_camel(k): v for k, v in snake_case_dict.items()}
+
+@pytest.fixture
+def event_data_camel_case(event_data):
+    return [convert_dict_keys_to_camel_case(data) for data in event_data]
+
+def test_write_all_events_to_fabric(
+        event_replicator: EventsReplicator, 
+        event_data_camel_case: List[dict], 
+        mock_get_events: Any, 
+        mock_set_event_state: Any, 
+        mock_write_events_to_lakehouse_tables: Any, 
+        mocker):
+    mocker.patch("cdf_fabric_replicator.event.EventsReplicator.get_event_state", return_value=None)
+    event_replicator.process_events()
+    mock_get_events.assert_called_with(event_replicator.config.event.batch_size, 0)
+    mock_write_events_to_lakehouse_tables.assert_called_once()
+    mock_write_events_to_lakehouse_tables.assert_called_with(
+        event_data_camel_case, 
+        event_replicator.config.event.lakehouse_abfss_path_events
+    )
+    mock_set_event_state.assert_called_with(
+        event_replicator.event_state_key, 
+        event_data_camel_case[-1]["createdTime"]
+    )
```

Comment:
These test might be easier to read if has Arrange Act Assert sections in each one

Reply:
I'll add the sections for these unit tests since they are applicable here. However, please note that sometimes Pytest may not have the "arrange" section as they sometimes are taken care of by fixtures.